File: mobile_app/lib/models/ecg_record_model.dart
class EcgRecord {
  int id;
  int userId;
  int deviceId;
  String dataDirectory;

  EcgRecord({
    required this.id,
    required this.userId,
    required this.deviceId,
    required this.dataDirectory,
  });

  factory EcgRecord.fromJson(Map<String, dynamic> json) => EcgRecord(
    id: json["id"],
    userId: json["user_id"],
    deviceId: json["device_id"],
    dataDirectory: json["data_directory"],
  );

  Map<String, dynamic> toJson() => {
    "id": id,
    "user_id": userId,
    "device_id": deviceId,
    "data_directory": dataDirectory,
  };
}

File: server/Models/ecgRecordModel.js
const Sequelize = require('sequelize');
const sequelize = require('../util/db');
const User = require('./userModel');

const EcgRecords = sequelize.define('ecg_record', {
  record_id: {
    type: Sequelize.INTEGER,
    autoIncrement: true,
    allowNull: false,
    primaryKey: true
  },
  user_id: {
    type: Sequelize.INTEGER,
    allowNull: false,
    references: {
      model: User,
      key: 'user_id'
    }
  },
  device_id: {
    type: Sequelize.STRING,
    allowNull: false
  },
  data_directory: {
    type: Sequelize.STRING,
    allowNull: false
  },
  start_time: {
    type: Sequelize.DATE,
    allowNull: false
  },
  stop_time: {
    type: Sequelize.DATE,
    allowNull: false
  },
  sensor_type: {
    type: Sequelize.STRING,
    allowNull: false
  },
},
{
  updatedAt: 'updated_at',
  createdAt: 'created_at',
  tableName: 'ecg_records'
});

User.hasMany(EcgRecords, { foreignKey: 'user_id' });
EcgRecords.belongsTo(User, { foreignKey: 'user_id' });

module.exports = EcgRecords;

